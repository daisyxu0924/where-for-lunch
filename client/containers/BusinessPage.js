import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Link, withRouter } from 'react-router-dom';
import Button from 'components/Button/Button';
import placeActions from 'actions/placeActions';
import Place from 'components/Place/Place';
import Photo from '../components/Photo/Photo';
import PlaceInfo from '../components/PlaceInfo/PlaceInfo';
import { newWindow } from '../lib/navigatorHelper';

class BusinessPage extends Component {
  handleOnClickShowMap = () => {
  }
  handleOnClickGotoLink = () => {
    newWindow(this.props.place.url);
  }
  handleOnClickBack = () => {
    this.props.history.goBack();
  }
  componentDidMount = () => {
    this.props.fetchPlacedetails(this.props.place.id || this.props.match.params.id);
    window.scrollTo(0, 200); // added this for u to see the changes that there add at the bottom
  }
  render() {
    const { place } = this.props;
    return (
      <div className="homePageWrapper">
        <Place place={place} />
        <Link to={`detail/${place.id}`}>
          <Button onClick={this.handleOnClickBack} theme="BusinessPage" icon={'back'} title='Back' />
        </Link>
        <Button onClick={this.handleOnClickGotoLink} theme="BusinessPage" icon={'page'} title='Visit Website' />
        <Button onClick={this.handleOnClickShowMap} theme="BusinessPage" icon={'map'} title='Show Map' />
        <PlaceInfo place={place} />
        <Photo place={place} />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  condition: state.condition,
  place: state.place,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    fetchPlacedetails: placeActions.fetchPlacedetails,
  }, dispatch);

BusinessPage.propTypes = {
  place: PropTypes.object,
  match: PropTypes.object,
  history: PropTypes.object,
  fetchPlacedetails: PropTypes.func,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withRouter(BusinessPage));
