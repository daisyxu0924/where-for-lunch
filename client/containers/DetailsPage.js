import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import placeActions from 'actions/placeActions';
import Place from 'components/Place/Place';
import Gallery from 'components/Gallery/Gallery';

class DetailsPage extends Component {
  componentDidMount() {
    this.props.fetchPlace(this.props.match.params.id);
  }

  render() {
    const { place } = this.props;

    return (
      <div className="detailsPageWrapper">
        <Place place={place} />
        <Gallery photos={place.photos} />
      </div>
    );
  }
}

DetailsPage.propTypes = {
  place: PropTypes.shape({
    name: PropTypes.string,
    address: PropTypes.string,
    phone: PropTypes.string,
    price: PropTypes.string,
    rating: PropTypes.number,
    reviewCount: PropTypes.number,
    categories: PropTypes.arrayOf(PropTypes.string),
    photos: PropTypes.arrayOf(PropTypes.string),
  }).isRequired,
  fetchPlace: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
    }),
  }),
};

DetailsPage.defaultProps = {
  place: {
    name: '',
    address: null,
    phone: null,
    price: null,
    rating: null,
    reviewCount: null,
    categories: null,
    photos: null,
  },
};

const mapStateToProps = state => ({
  place: state.place,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchPlace: placeActions.fetchPlace,
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(DetailsPage);
