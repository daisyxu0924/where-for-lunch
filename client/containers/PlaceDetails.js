import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import placeActions from 'actions/placeActions';
import PlaceDetail from 'components/PlaceDetail/PlaceDetail';

class PlaceDetails extends Component {
  constructor(props) {
    super(props);

    if (this.props.match) {
      const id = this.props.match.params.id;

      this.props.fetchPlaceDetails(id);
    }
  }

  render() {
    const { details } = this.props.place;
    return (
      <div className="placeDetailsWrapper">
        { details ? <PlaceDetail details={details}/> : 'Loading, please wait'}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  place: state.place,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    fetchPlaceDetails: placeActions.fetchPlaceDetails,
  }, dispatch);

PlaceDetails.propTypes = {
  match: PropTypes.object,
  place: PropTypes.object,
  fetchPlaceDetails: PropTypes.func,
};
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PlaceDetails);
