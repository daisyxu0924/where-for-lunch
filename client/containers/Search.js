import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import Button from 'components/Button/Button';
import Condition from 'components/Condition/Condition';
import conditionActions from 'actions/conditionActions';
import placeActions from 'actions/placeActions';

export class Search extends PureComponent {
  static propTypes = {
    condition: PropTypes.object,
    place: PropTypes.object,
    setRadius: PropTypes.func.isRequired,
    fetchPlaces: PropTypes.func.isRequired,
    setPrices: PropTypes.func.isRequired,
  };

  handleOnPriceClick = (value) => {
    this.props.setPrices(value);
  }

  handleOnConditionChange = (value) => {
    this.props.setRadius(value);
  }

  handleButtonClick = () => {
    const { prices, ...newCondition } = this.props.condition;
    this.props.fetchPlaces({
      ...newCondition,
      price: prices.toString(),
    });
  };

  render() {
    const { condition } = this.props;
    const disableBtn = !(condition.latitude && condition.longitude);

    return (
      <div className="searchWrapper">
        <Condition
          condition={condition}
          action={this.handleOnConditionChange}
          onClickPrice={this.handleOnPriceClick}
        />
        <Button
          theme="homepageClick"
          onClick={this.handleButtonClick}
          disabled={disableBtn}
        />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  condition: state.condition,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    setRadius: conditionActions.setRadius,
    fetchPlaces: placeActions.fetchPlaces,
    setPrices: conditionActions.setPrices,
  }, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Search);
