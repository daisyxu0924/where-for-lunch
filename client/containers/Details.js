import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import placeActions from 'actions/placeActions';
import PlaceDetails from 'components/PlaceDetails/PlaceDetails';

export class Details extends PureComponent {
  static defaultProps = {
    place: {},
  };

  static propTypes = {
    fetchDetails: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
    match: PropTypes.object.isRequired,
    setDetails: PropTypes.func.isRequired,
    place: PropTypes.object,
  };

  state = {
    hasError: false,
  };

  componentDidMount() {
    const { fetchDetails, match: { params } } = this.props;
    fetchDetails(params.placeid);
  }

  render() {
    if (this.props.loading) {
      return (
        <span>Loading...</span>
      );
    }

    return (
      <PlaceDetails place={this.props.place} />
    );
  }
}

const mapStateToProps = state => ({
  place: state.place.details,
  loading: state.place.loading,
});

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    setDetails: placeActions.setDetails,
    fetchDetails: placeActions.fetchDetails,
  }, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Details);
