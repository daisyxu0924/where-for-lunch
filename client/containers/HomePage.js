import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import Place from 'components/Place/Place';
import PlaceLink from 'components/PlaceLink/PlaceLink';
import Search from 'containers/Search';
import Details from 'containers/Details';

export class HomePage extends PureComponent {
  static propTypes = {
    place: PropTypes.object,
    match: PropTypes.object,
  };

  render() {
    const { place } = this.props;
    const PlaceComponent = place.id ?
      <PlaceLink place={place} /> :
      <Place place={place} />;

    return (
      <div className="homePageWrapper">
        <Route path="/" exact component={ () => PlaceComponent } />
        <Route path="/" exact component={ Search } />

        <Route path="/details/:placeid" exact component={Details} />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  place: state.place,
});

export default connect(
  mapStateToProps,
)(HomePage);
