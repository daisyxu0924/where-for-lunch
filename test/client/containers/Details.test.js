import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Details } from 'containers/Details';

describe('Search Container', () => {
  const mockFn = jest.fn();
  const defaultProps = {
    fetchDetails: mockFn,
    loading: false,
    match: {
      params: {
        placeid: '',
      },
    },
    setDetails: mockFn,
    place: {},
  };
  const detailsComponent = props =>
    <Details {...{ ...defaultProps, ...props }}/>;

  test('renders `Loading...` text if props.loading = true', () => {
    const wrapper = shallow(detailsComponent({
      loading: true,
    }));

    expect(wrapper.find('span').first().text()).toBe('Loading...');
    expect(wrapper.find('PlaceDetails').length).toEqual(0);
  });

  test('renders `PlaceDetails` if props.loading = false', () => {
    const wrapper = shallow(detailsComponent());

    expect(wrapper.find('span').length).toEqual(0);
    expect(wrapper.find('PlaceDetails').length).toEqual(1);
  });

  test('renders properly', () => {
    const wrapper = renderer
      .create(detailsComponent())
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
