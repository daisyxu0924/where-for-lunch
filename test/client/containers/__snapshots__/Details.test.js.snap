// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Container renders properly 1`] = `
<section
  className="root"
>
  <header
    className="header"
  >
    <h1>
      <span
        className="badge green"
      >
        OPEN
      </span>
    </h1>
    <span />
    <p />
  </header>
  <div
    className="row"
  >
    <div
      className="contactDetails"
    >
      <p>
        Address:Â 
      </p>
      <div
        className="root"
      >
        <div
          hidden={true}
          id="map"
        />
      </div>
    </div>
  </div>
</section>
`;
